letsencrypt/cli.py

```diff
@@ -693,6 +706,94 @@ def install(args, config, plugins):
     le_client.enhance_config(domains, config)
 
 
+def renew(args, cli_config, plugins):
+    """Renew previously-obtained certificates."""
+    cli_config = configuration.RenewerConfiguration(cli_config)
+    configs_dir = cli_config.renewal_configs_dir
+    for renewal_file in reversed(os.listdir(configs_dir)):
+        if not renewal_file.endswith(".conf"):
+            continue
+        print("Processing " + renewal_file)
+        # XXX: does this succeed in making a fully independent config object
+        #      each time?
+        config = configuration.RenewerConfiguration(copy.deepcopy(cli_config))
+        full_path = os.path.join(configs_dir, renewal_file)
+        try:
+            renewal_candidate = storage.RenewableCert(full_path, config)
+        except (errors.CertStorageError, IOError):
+            logger.warning("Renewal configuration file %s is broken. "
+                           "Skipping.", full_path)
+            continue
+        print(renewal_candidate.names(), renewal_candidate.should_autorenew())
+        if "renewalparams" not in renewal_candidate.configuration:
+            logger.warning("Renewal configuration file %s lacks "
+                           "renewalparams. Skipping.", full_path)
+            continue
+        renewalparams = renewal_candidate.configuration["renewalparams"]
+        if "authenticator" not in renewalparams:
+            logger.warning("Renewal configuration file %s does not specify "
+                           "an authenticator. Skipping.", full_path)
+            continue
+    # ?? config = configuration.NamespaceConfig(_AttrDict(renewalparams))
+        # XXX: also need: webroot_map
+        # XXX: also need: nginx_, apache_, and plesk_ items
+        # string-valued items to add if they're present
+        for config_item in ["config_dir", "log_dir", "work_dir", "user_agent",
```

Comment:
set these lists as constants outside this function (though in this file is fine).